Source/WebCore/PAL/pal/text/TextCodec.cpp

```diff
@@ -49,14 +49,20 @@ int TextCodec::getUnencodableReplacement(char32_t codePoint, UnencodableHandling
         codePoint = replacementCharacter;
 
     switch (handling) {
-    case UnencodableHandling::Entities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
-    case UnencodableHandling::URLEncodedEntities:
-        return snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "%%26%%23%u%%3B", static_cast<unsigned>(codePoint));
+    case UnencodableHandling::Entities: {
+        int count = snprintf(replacement.data(), sizeof(UnencodableReplacementArray), "&#%u;", static_cast<unsigned>(codePoint));
+        ASSERT(count >= 0);
+        return std::span { replacement }.first(static_cast<size_t>(std::max<int>(0, count)));
```

Comment:
The `<int>` here is not needed since both `0` and `count` are `int`.

Is the cast to `size_t` needed? I'd think this would work the same without it.

Reply:
Fixed